### PR TITLE
add default seccomp profile as json

### PR DESCRIPTION
**Pull request by jessfraz**
profile is created by go generate

related to #20092

this way people can have an easy way to use the default whitelist as a base for their own custom profiles

### CODE REVIEW

**Comment by jfrazelle** [comment]
going to add a validator like validate-vendor

**Comment by thaJeztah** [comment]
LGTM

should we have something in place to make sure the JSON export does not get out of sync with he one in code?

**Comment by jfrazelle** [comment]
ya thats what i was thinking with the validate script, working on it :)

**Comment by jfrazelle** [comment]
added ;)

File: profiles/seccomp/generate.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	// get the default profile
+	p := seccomp.GetDefaultProfile()
+
+	// write the default profile to the file
+	b, err := json.MarshalIndent(p, "", "\t")
+	if err != nil {
+		panic(err)
+	}
+
+	if err := ioutil.WriteFile(f, b, 0755); err != nil {
```
Comment (calavera) [inline profiles/seccomp/generate.go]: s/0755/0644/ ? that json file is not going to execute itself any time soon :smile_cat:

Comment (jfrazelle) [inline profiles/seccomp/generate.go]: lolz

**Comment by calavera** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM, thanks Jess!

**Comment by runcom** [comment]
I'm getting:
```sh
$ docker run --rm --security-opt seccomp:`pwd`/profiles/seccomp/default.json -i debian:jessie \
    unshare --map-root-user --user sh -c whoami

docker: Error response from daemon: Cannot start container 099c2b60a0cda168811abe796c4ccc4c83f90a33aaa13a5ace456fb16b89c002: Decoding seccomp profile failed: json: cannot unmarshal number into Go value of type types.Action.
```
cause seems like the exported json is using libcontainer/configs types - instead the --security-opt flag wants a json with engine-api types. Am I doing something wrong here?

/cc @thaJeztah

**Comment by jfrazelle** [comment]
ya we need to convert to the engine-api type i can make a patch

**Comment by tianon** [comment]
:metal:

**Pull request merged**